src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -525,51 +532,79 @@ internal static void EncodeInternal(TypeSymbol type, int customModifiersCount, R
                     transformFlagsBuilder.Add(false);
                 }
 
-                // Native compiler encodes an extra transform flag, always false, for each custom modifier.
-                HandleCustomModifiers(customModifiersCount, transformFlagsBuilder);
-
-                type.VisitType(s_encodeDynamicTransform, transformFlagsBuilder);
+                if (addCustomModifierFlags)
+                {
+                    // Native compiler encodes an extra transform flag, always false, for each custom modifier.
+                    HandleCustomModifiers(customModifiersCount, transformFlagsBuilder);
+                    type.VisitType(Transformer.WithCustomModifierFlags, transformFlagsBuilder);
+                }
+                else
+                {
+                    type.VisitType(Transformer.WithoutCustomModifierFlags, transformFlagsBuilder);
+                }
             }
 
-            private static readonly Func<TypeSymbol, ArrayBuilder<bool>, bool, bool> s_encodeDynamicTransform = (type, transformFlagsBuilder, isNestedNamedType) =>
+            private class Transformer
```

Comment:
Is this class necessary or could the two uses be above be replaced by ```type.VisitType((type, builder, isNested) => AddFlags(type, builder, isNested, true))``` and ```type.VisitType(..., false))```, where `AddFlags` is a static method?